File: rb/features.rb
# Imgur SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ImgurFeatures
  def self.make_feature(name)
    case name
    when "base"
      ImgurBaseFeature.new
    when "ratelimit"
      ImgurRatelimitFeature.new
    when "retry"
      ImgurRetryFeature.new
    when "test"
      ImgurTestFeature.new
    when "timeout"
      ImgurTimeoutFeature.new
    else
      ImgurBaseFeature.new
    end
  end
end

File: py/features.py
# Imgur SDK feature factory

from feature.base_feature import ImgurBaseFeature
from feature.test_feature import ImgurTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ImgurBaseFeature(),
        "test": lambda: ImgurTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
